Civi/Core/Service/AutoSubscriber.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+namespace Civi\Core\Service;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * AutoSubscriber allows child classes to listen to events.
+ *
+ * Child classes must implement the `getSubscribedEvents` method, and the callbacks
+ * it returns will be automatically registered.
+ *
+ * @service
+ * @internal
```

Comment:
Conceptually, isn't this saying the same thing as the `AutoDefinition.php` patch (`is_a(...AutoSubscriber::class...)`)?

My guess is that these lines are more illustrative and less functional? In which case, maybe say something like:

```
* This class implies @service @internal on all sub-classes.
```